sql/localsql/nipost/poet_registration.go

```diff
@@ -37,18 +38,48 @@ func AddPoetRegistration(
 	return nil
 }
 
-func PoetRegistrationCount(db sql.Executor, nodeID types.NodeID) (int, error) {
-	var count int
-	enc := func(stmt *sql.Statement) {
-		stmt.BindBytes(1, nodeID.Bytes())
+func PoetRegistrationCount(db sql.Executor, nodeID types.NodeID, addresses ...string) (int, error) {
+	var (
+		count int
+		enc   func(stmt *sql.Statement)
+		query string
+	)
+
+	if len(addresses) > 0 {
+		enc = func(stmt *sql.Statement) {
+			stmt.BindBytes(1, nodeID.Bytes())
+			for i, addr := range addresses {
+				stmt.BindText(i+2, addr) // Start binding addresses at position 2
+			}
+		}
+
+		placeholders := make([]string, len(addresses))
+		for i := range placeholders {
+			placeholders[i] = fmt.Sprintf("?%d", i+2) // Generate placeholders starting from ?2
+		}
+		// Construct the query with IN clause for addresses
+		query = fmt.Sprintf(`select count(*) from poet_registration where id = ?1 and address IN (%s);`,
+			strings.Join(placeholders, ", "))
+
+	} else {
+		enc = func(stmt *sql.Statement) {
+			stmt.BindBytes(1, nodeID.Bytes())
+		}
+
+		query = `select count(*) from poet_registration where id = ?1;`
 	}
+
```

Comment:
This could adopt the `sql/builder` and pass the filter of addresses accordingly - see for instance `ActivationsCount` in the `api/grpcserver/v2alpha1` package for an example that filters with such a builder. @kacpersaw could maybe also help with adapting it 🙂

Reply:
I think that querying for the _count_ of registrations might not be needed in general, query for registrations should be sufficient. There, another option, perhaps simpler, could be to query one by one and collect into a slice. Wdyt?

Reply:
Sure that would work too; just want to avoid the functions in `sql` from becoming too complex - better 2 simple functions than one that does multiple things 🙂

Reply:
made 2 separate methods. 
By the way: why do not we use ORM and instead of it our own sql package?